src/FileServiceManager/FileService/DAL/FileSystemConfiguration/FileSystemJsonConfiguration/FileSystemJsonConfigurator.cs

```diff
@@ -0,0 +1,229 @@
+                        }
+                    }
+                }
+                else
+                {
+                    Log.Error("FileSystemConfig.json is not found");
+                }
+            }
+
+            catch (Exception ex)
+            {
+                Log.Error(ex.Message);
+            }
+            finally
+            {
+                fileReader?.Close();
+                fileWriter?.Close();
+            }
+        }
+
+        public async IAsyncEnumerable<FileMetadata> GetAllConfigurationsAsync(int companyId)
+        {
+            JToken rootToken = null;
+            StreamReader file = null;
+            JToken filesToken = null;
+            try
+            {
+                using (file = File.OpenText(_configPath))
+                {
+                    using (JsonTextReader reader = new JsonTextReader(file))
+                    {
+                        rootToken = await JToken.ReadFromAsync(reader);
+                    }
+                }
+
+                if (rootToken != null)
+                {
+                    filesToken = rootToken.SelectToken($"$.companies[?(@.companyId == {companyId})]")?.SelectToken($"$..files");
+                }
+                else
+                {
+                    Log.Error("FileSystemConfig.json is not found");
+                }
+            }
+
+            catch (Exception ex)
+            {
+                Log.Error(ex.Message);
+            }
+
+            finally
+            {
+                file?.Close();
+            }
+
+            if (filesToken == null)
+            {
+                Log.Error($"No file for company ID = {companyId}");
+            }
+            else
+            {
+                foreach (var fileToken in filesToken)
+                {
+                    yield return fileToken.ToObject<FileMetadata>();
+                }
+            }
+        }
+
+        public async Task<FileMetadata> GetConfigurationAsync(int companyId, Guid fileId)
+        {
+            JToken rootToken = null;
+            StreamReader file = null;
+            try
+            {
+                using (file = File.OpenText(_configPath))
+                {
+                    using (JsonTextReader reader = new JsonTextReader(file))
+                    {
+                        rootToken = await JToken.ReadFromAsync(reader);
+                    }
+                }
+
+                if (rootToken != null)
+                {
+                    JToken filesToken = rootToken.SelectToken($"$.companies[?(@.companyId == {companyId})]")?.SelectToken($"$..files[?(@.Id == '{fileId}')]");
+
+                    if (filesToken != null)
+                    {
+                        return filesToken.ToObject<FileMetadata>();
+                    }
+                    else
+                    {
+                        Log.Error($"No file with id = {fileId} for company ID = {companyId}");
```

Comment:
in this case we need to pass to client not found or something like that

Reply:
+